testing/mock/ibc_module.go

```diff
@@ -14,6 +14,13 @@ import (
 	"github.com/cosmos/ibc-go/v7/modules/core/exported"
 )
 
+// applicationCallbackError is a custom error type that will be unique for testing purposes.
+type applicationCallbackError struct{}
```

Comment:
This was added because there used to be a testcase that checked the exact error type returned by the `OnChanUpgradeRestore` application callback. However that function no longer returns an error.

The idea behind this was that we can do `suite.Require().ErrorIs(err)` to assert a custom error type with `fmt.Errorf`, it would just be a regular go error and we could run into false positives. 

While not used in this PR, I think we could replace the usage of our failed application callback test cases with this. WDYT?

Reply:
oh, nevermind, I see its exposed with assignment to `MockApplicationCallbackError` in mock.go 👍